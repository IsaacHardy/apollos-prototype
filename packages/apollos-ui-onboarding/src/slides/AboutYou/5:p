import React, { memo } from 'react';
import { Query } from 'react-apollo';
import PropTypes from 'prop-types';
// This query is also found in core/permissionUtils. We should refactor into a notifications module.
import { GET_NOTIFICATIONS_ENABLED } from '@apollosproject/ui-notifications';

import AskNotifications from './AskNotifications';
// eslint-disable-next-line react/display-name
const AskNotificationsConnected = memo(
  ({ Component, onSwipe, onRequestPushPermissions, onCompleted, ...props }) => {
    const complete = async () => {
      await onCompleted();
      onSwipe();
    };
    return (
      <Query query={GET_NOTIFICATIONS_ENABLED}>
        {({ data: { notificationsEnabled = false } = {} }) => (
          <Component
            onPressButton={() => onRequestPushPermissions()}
            buttonDisabled={notificationsEnabled}
            buttonText={
              notificationsEnabled
                ? 'Notifications Enabled!'
                : 'Yes, enable notifications'
            }
            // next button
            onSwipe={notificationsEnabled ? complete : null}
            // skip button
            onPressSecondary={notificationsEnabled ? null : onSwipe}
            pressPrimaryEventName={'Ask Notifications Completed'}
            pressSecondaryEventName={'Ask Notifications Skipped'}
            {...props}
          />
        )}
      </Query>
    );
  }
);

AskNotificationsConnected.propTypes = {
  Component: PropTypes.node, // Custom component to be rendered. Defaults to AskNotifications
  onSwipe: PropTypes.func,
  onRequestPushPermissions: PropTypes.func.isRequired,
  onCompleted: PropTypes.func,
};

AskNotificationsConnected.defaultProps = {
  Component: AskNotifications,
};

AskNotificationsConnected.displayName = 'AskNotificationsConnected';

export default AskNotificationsConnected;
