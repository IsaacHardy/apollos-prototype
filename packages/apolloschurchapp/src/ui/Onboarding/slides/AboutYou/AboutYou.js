import React, { memo } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { get } from 'lodash';

import {
  styled,
  H2,
  H5,
  Radio,
  RadioButton,
  H6,
  DateInput,
  PaddedView,
} from '@apollosproject/ui-kit';

import Slide from '../../Slide';

const Title = styled(({ theme }) => ({
  color: theme.colors.primary,
  marginBottom: theme.sizing.baseUnit * 0.5,
}))(H2);

const Description = styled(({ theme }) => ({
  color: theme.colors.text.secondary,
  marginBottom: theme.sizing.baseUnit,
}))(H5);

const Label = styled({
  color: 'gray',
  opacity: 0.7,
})(H6);

const StyledDate = styled(({ theme }) => ({
  marginTop: 0,
  marginBottom: theme.sizing.baseUnit,
}))(DateInput);

const StyledRadio = styled(({ theme }) => ({
  marginBottom: theme.sizing.baseUnit,
  flexDirection: 'row',
}))(Radio);

const RadioLabel = styled(({ theme }) => ({
  marginLeft: theme.sizing.baseUnit * 0.5,
}))(H5);

const AboutYou = memo(
  ({
    slideTitle,
    description,
    defaultDate,
    genderList,
    values,
    touched,
    errors,
    setFieldValue,
    BackgroundComponent,
    ...props
  }) => (
    <Slide {...props}>
      {BackgroundComponent}
      <PaddedView>
        <Title>{slideTitle}</Title>
        <Description>{description}</Description>
        <Label>Gender</Label>
        <StyledRadio
          label="Gender"
          type="radio"
          value={get(values, 'gender')}
          error={get(touched, 'gender') && get(errors, 'gender')}
          onChange={(value) => setFieldValue('gender', value)}
        >
          {genderList.map((gender) => [
            <RadioButton
              key={gender}
              value={gender}
              label={() => <RadioLabel>{gender}</RadioLabel>}
              underline={false}
            />,
          ])}
        </StyledRadio>
        <Label>Birthday</Label>
        <StyledDate
          type={'DateInput'}
          placeholder={'Select a date...'}
          value={moment
            .utc(get(values, 'birthDate', defaultDate) || defaultDate)
            .toDate()}
          error={get(touched, 'birthDate') && get(errors, 'birthDate')}
          displayValue={moment(
            get(values, 'birthDate', defaultDate) || defaultDate
          ).format('MM/DD/YYYY')}
          onChange={(value) => setFieldValue('birthDate', value)}
        />
      </PaddedView>
    </Slide>
  )
);

AboutYou.propTypes = {
  /* The `Swiper` component used in `<Onboading>` looks for and hijacks the title prop of it's
   * children. Thus we have to use more unique name.
   */
  slideTitle: PropTypes.string,
  description: PropTypes.string,
  defaultDate: PropTypes.instanceOf(Date),
  genderList: PropTypes.arrayOf(PropTypes.number),
  values: PropTypes.shape({}),
  touched: PropTypes.shape({}),
  errors: PropTypes.shape({}),
  setFieldValue: PropTypes.func.isRequired,
  /* Recommended usage:
   * - `Image` (react-native)
   * - `GradientOverlayImage` (@apollosproject/ui-kit) for increased readability
   * - `Video` (react-native-video) because moving pictures!
   */
  BackgroundComponent: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

AboutYou.defaultProps = {
  slideTitle: "This one's easy.",
  description:
    'Help us understand who you are so we can connect you with the best ministries and events.',
  genderList: ['Male', 'Female'],
  defaultDate: new Date(),
};

AboutYou.displayName = 'AboutYou';

export default AboutYou;
