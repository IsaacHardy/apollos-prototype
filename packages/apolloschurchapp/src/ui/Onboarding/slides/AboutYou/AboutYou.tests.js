import React from 'react';
import renderer from 'react-test-renderer';
import { GradientOverlayImage } from '@apollosproject/ui-kit';

import Providers from 'apolloschurchapp/src/Providers';

import AboutYou from '.';

describe('The Onboarding AboutYou component', () => {
  it('should render', () => {
    /* we have to pass in a date via defaultDate or the DateInput component will create a current date
     * object and invalidate the snapshots every time. */
    const tree = renderer.create(
      <Providers>
        <AboutYou defaultDate={'2019-02-14'} setFieldValue={jest.fn()} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
