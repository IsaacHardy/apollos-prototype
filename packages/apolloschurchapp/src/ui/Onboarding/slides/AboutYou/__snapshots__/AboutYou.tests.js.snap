// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Onboarding AboutYou component should render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "marginBottom": 48,
              "paddingBottom": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 20,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "paddingBottom": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#00676D",
                    "fontFamily": "InterUI-Black",
                    "fontSize": 36,
                    "lineHeight": 41.4,
                  }
                }
              >
                This one's easy.
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 20.16,
                }
              }
            >
              Help us understand who you are so we can connect you with the best ministries and events.
            </Text>
          </View>
          <Text
            padded={true}
            style={
              Object {
                "color": "gray",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "marginTop": 16,
                "opacity": 0.7,
                "paddingBottom": 8.64,
              }
            }
          >
            Gender
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 16,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "rgba(165, 165, 165, 0.30000000000000004)",
                  "borderBottomWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 8,
                }
              }
              underline={false}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#A5A5A5",
                    "borderRadius": 99,
                    "borderWidth": 2,
                    "height": 20,
                    "justifyContent": "center",
                    "width": 20,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "marginLeft": 8,
                  }
                }
              >
                Male
              </Text>
            </View>
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "rgba(165, 165, 165, 0.30000000000000004)",
                  "borderBottomWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 8,
                }
              }
              underline={false}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#A5A5A5",
                    "borderRadius": 99,
                    "borderWidth": 2,
                    "height": 20,
                    "justifyContent": "center",
                    "width": 20,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "fontFamily": "InterUI-Medium",
                    "fontSize": 14,
                    "lineHeight": 20.16,
                    "marginLeft": 8,
                  }
                }
              >
                Female
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "gray",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "opacity": 0.7,
              }
            }
          >
            Birthday
          </Text>
          <View
            style={
              Object {
                "marginVertical": 16,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                bordered={false}
                disabled={false}
                pill={false}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#A5A5A5",
                    "borderColor": "#A5A5A5",
                    "borderRadius": 16,
                    "borderWidth": 2,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 32,
                    "justifyContent": "center",
                    "marginBottom": 8,
                    "marginRight": 8,
                    "marginTop": 5,
                    "opacity": 1,
                    "overflow": "hidden",
                    "paddingHorizontal": 4,
                    "paddingVertical": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignItems": "center",
                      "color": "#303030",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 12,
                      "justifyContent": "center",
                      "lineHeight": 17.28,
                      "paddingHorizontal": 6,
                      "textAlign": "center",
                    }
                  }
                >
                  Select a date...
                </Text>
              </View>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              hideModalContentWhileAnimating={false}
              onModalHide={[Function]}
              onRequestClose={[Function]}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "black",
                    "bottom": 0,
                    "height": 1334,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 750,
                  }
                }
              />
              <View
                hideModalContentWhileAnimating={false}
                onModalHide={[Function]}
                pointerEvents="box-none"
                scrollOffset={0}
                scrollOffsetMax={0}
                scrollTo={null}
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "flex-end",
                    "margin": 10,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
                supportedOrientations={
                  Array [
                    "portrait",
                    "landscape",
                  ]
                }
                swipeThreshold={100}
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 13,
                        "marginBottom": 8,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderBottomColor": "#d5d5d5",
                        "borderBottomWidth": 0.5,
                        "padding": 14,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#8f8f8f",
                            "fontSize": 13,
                            "textAlign": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      Pick a date
                    </Text>
                  </View>
                  <View
                    onStartShouldSetResponderCapture={[Function]}
                  >
                    <DatePicker
                      date={2019-02-14T00:00:00.000Z}
                      minuteInterval={1}
                      mode="date"
                      onCancel={[Function]}
                      onConfirm={[Function]}
                      onDateChange={[Function]}
                      onHideAfterConfirm={[Function]}
                    />
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "#d5d5d5",
                        "borderTopWidth": 0.5,
                        "height": 57,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#007ff9",
                            "fontSize": 20,
                            "fontWeight": "normal",
                            "textAlign": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 13,
                      "height": 57,
                      "justifyContent": "center",
                      "marginBottom": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#007ff9",
                          "fontSize": 20,
                          "fontWeight": "600",
                          "padding": 10,
                          "textAlign": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </Modal>
            <View
              onLayout={[Function]}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "justifyContent": "center",
                  "opacity": 0.8,
                  "position": "absolute",
                  "top": 0,
                  "transform": Array [
                    Object {
                      "scale": 0.8,
                    },
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#505050",
                    "fontFamily": "InterUI-Bold",
                    "fontSize": 12,
                    "lineHeight": 17.28,
                    "paddingVertical": 4,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
